### PR TITLE
Create `/itemname` command that names held items.

### PR DESCRIPTION
**Pull request by zsturgess**
Whilst peeking at the EssentialsX repo for an update on 1.14, I noticed a couple of PRs that are marked for v2.17 that seem to have been abandoned by their original authors. I figured adopting them and making basic edits from review comments wasn't beyond my otherwise rather basic java abilities.

This original PR, from @SupaHam, is at https://github.com/EssentialsX/Essentials/pull/672

My comments:
- @md678685 's review originally suggested altering the error message to `You need to hold an item to rename it.` (and renaming the key). I'm in no position to really judge if there are un-renamable materials or not, so my compromise wording (`You need to hold a renamable item to rename it.`) is included in this PR 
- @md678685 's review originally suggested adding `/rename` as an alias. Personally, I think that's too general a term and we may live to regret using such a generic word for a command, so I've added `/itemrename`, `/irename` and the e-prefixed variants instead.

If this sort of PR adoption is welcome, please let it be known and I might just take on a few more in my spare time.

### CODE REVIEW

File: Essentials/src/messages.properties [hunk 1]
```diff
@@ -625,3 +625,64 @@ year=year
...
+spectator=spectator
+kitContains=\u00a76Kit \u00a7c{0} \u00a76contains:
+kitItem=\u00a76- \u00a7f{0}
+invalidBanner=\u00a74Invalid banner syntax.
+msgDisabled=\u00a76Receiving messages \u00a7cdisabled\u00a76.
+msgDisabledFor=\u00a76Receiving messages \u00a7cdisabled \u00a76for \u00a7c{0}\u00a76.
+msgEnabled=\u00a76Receiving messages \u00a7cenabled\u00a76.
+msgEnabledFor=\u00a76Receiving messages \u00a7cenabled \u00a76for \u00a7c{0}\u00a76.
+msgIgnore=\u00a7c{0} \u00a74has messages disabled.
+minimumPayAmount=\u00a7cThe minimum amount you can pay is {0}.
```
Comment (mdcfe) [inline Essentials/src/messages.properties]: These extra messages shouldn't be in this PR.

File: Essentials/src/messages.properties [hunk 2]
```diff
@@ -625,3 +625,64 @@ year=year
...
+itemnameInvalidItem=\u00a7cYou need to hold a renamable item to rename it.
```
Comment (mdcfe) [inline Essentials/src/messages.properties]: The only material that this command will refuse to rename is air (an empty hand). I maintain that my earlier suggestion is more logical, as an empty hand is not an item to the majority of players.

File: Essentials/src/plugin.yml [hunk 3]
```diff
@@ -208,6 +208,10 @@ commands:
     description: Searches for an item.
     usage: /<command> <item>
     aliases: [dura,edura,durability,edurability,eitemdb,itemno,eitemno]
+  itemname:
+    description: Names an item.
+    usage: /<command> <name...>
```
Comment (mdcfe) [inline Essentials/src/plugin.yml]: In line with other commands like `broadcast` and `ban`, this should be `<name>` without ellipsis.

**Review by mdcfe — CHANGES_REQUESTED** [review]
Some minor feedback on the command. I also think this command should let people clear the name, either with a `/itemname clear` subcommand or by simply not providing any arguments.

File: Essentials/src/plugin.yml (hunk 3 above)
Comment (zsturgess) [reply Essentials/src/plugin.yml]: Completed as requested.

File: Essentials/src/messages.properties (hunk 2 above)
Comment (zsturgess) [reply Essentials/src/messages.properties]: Understood - Apologies, I think I mis-read your suggestion on the original PR as a question :)
I will make this change as requested.

File: Essentials/src/messages.properties (hunk 1 above)
Comment (zsturgess) [reply Essentials/src/messages.properties]: Removed from the file. Not sure where they came from?

**Comment by zsturgess** [comment]
> 
> 
> Some minor feedback on the command. I also think this command should let people clear the name, either with a `/itemname clear` subcommand or by simply not providing any arguments.

"Clear" could introduce conflict with trying to name something that, so 👍 to no-arguments. I think this would be beyond my abilities - would you be willing to accept this PR and open up an issue to enhance that later? (Or perhaps someone with more codebase knowledge could take it from here!)

**Comment by mdcfe** [comment]
Closing in favour of #2573.

**Pull request closed without merging**